web/src/components/Korrel8rPanel.tsx

```diff
@@ -14,104 +14,69 @@ import {
 import './korrel8rpanel.css';
 import { getNeighborsGraph } from '../korrel8r-client';
-import { useDispatch, useSelector } from 'react-redux';
-import { setQueryResponse, setQuery } from '../redux-actions';
-import { State } from '../redux-reducers';
 import { Korrel8rGraphNeighboursResponse } from '../korrel8r/query.types';
 import { LoadingTopology } from './LoadingTopology';
 import { useTranslation } from 'react-i18next';
-import { useBoolean } from '../hooks/useBoolean';
 import { CubesIcon, ExclamationCircleIcon } from '@patternfly/react-icons';
-import { usePluginAvailable } from '../hooks/usePluginAvailable';
+import { useURLState } from '../hooks/useURLState';
 
-interface Korrel8rPanelProps {
-  initialQueryString: string;
-}
-
-export default function Korrel8rPanel({ initialQueryString }: Korrel8rPanelProps) {
-  const [queryInputField, setQueryInputField] = React.useState(initialQueryString);
-  const [errorMessage, setErrorMessage] = React.useState('');
-  const [errorMessageTitle, setErrorMessageTitle] = React.useState('');
+export default function Korrel8rPanel() {
   const { t } = useTranslation('plugin__troubleshooting-panel-console-plugin');
 
-  // Start off loading on the first render since the usePluginAvailable hook is async and will start
-  // loading immediately
-  const [isLoading, , setLoadingTrue, setLoadingFalse] = useBoolean(true);
-  const [netobserveAvailable, netobserveAvailableLoading] = usePluginAvailable('netobserv-plugin');
-  const [loggingAvailable, loggingAvailableLoading] = usePluginAvailable('logging-view-plugin');
-
-  const dispatch = useDispatch();
-  const savedQuery: string = useSelector((state: State) => state.plugins?.tp?.get('query'));
-  const queryResponse: Korrel8rGraphNeighboursResponse = useSelector((state: State) =>
-    state.plugins?.tp?.get('queryResponse'),
-  );
-
-  if (!savedQuery && queryInputField) {
-    dispatch(setQuery(queryInputField));
-  }
-
-  const handleError = React.useCallback(
-    (error) => {
-      // eslint-disable-next-line no-console
-      console.error(error);
-      setLoadingFalse();
-      let displayError: string;
-      try {
-        displayError = JSON.parse(error.message).error;
-        setErrorMessageTitle(t('Korrel8r Error'));
-      } catch (e) {
-        displayError = error.message;
-        setErrorMessageTitle(t('Error Loading Data'));
-      }
-      setErrorMessage(displayError);
-    },
-    [setLoadingFalse, setErrorMessageTitle, setErrorMessage, t],
+  // State
+  const { korrel8rQueryFromURL: contextQuery } = useURLState();
+  const [query, setQuery] = React.useState(contextQuery);
+  const [result, setResult] = React.useState(
+    null as { graph?: Korrel8rGraphNeighboursResponse; message?: string; title?: string },
   );
 
+  // Re-loading the graph is triggered by setting result = null
   React.useEffect(() => {
-    if (!savedQuery || netobserveAvailableLoading || loggingAvailableLoading) {
+    if (result != null) {
```

Comment:
Javascript has a lot of weird and very specific oddities with comparisons especially with null. Both this comparison and the one in the topology function should use the triple equal checks to ensure that only null is matched, `!==` and `===` respectively